src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+                b_neighbors[i[m]] = ii + new_order[m]
+            end   
+            # I use splice instead of insert so I can insert array   
+            splice!(b_list, ii+1:ii, pts_to_add)
+            skip = true
+            num_skips = length(i)
+        end
+    end
+
+    return b_neighbors, b_list
+end
+
+"""
+    _flag_ent_exit(
+        ::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, 
+        poly_b, a_list, b_list, edges_a, edges_b
+    )::a_list, b_list
+
+    This function flags all the intersection points as either an 'entry' or 'exit' point.
+"""
+function _flag_ent_exit(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b, a_list, b_list, edges_a, edges_b)
+    # Put in ent exit flags for poly_a
+    status = false
+    for ii in eachindex(a_list)
+        if ii == 1
+            temp = within(edges_a[ii][1], poly_b)
+            status = !(temp[1])
+            continue
+        end
+        if a_list[ii].inter
+            a_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+
+    # Put in ent exit flags for poly_b
+    status = false
+    for ii in eachindex(b_list)
+        if ii == 1
+            temp = within(edges_b[ii][1], poly_a)
+            status = !(temp[1])
+            continue
+        end
+        if b_list[ii].inter
+            b_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+
+    return a_list, b_list
+end
+
+"""
+    _build_ab_list(
+        poly_a, poly_b
+    )::a_list, b_list, a_idx_list, intr_list, edges_a, edges_b
+
+    This function calls '_build_a_list', '_build_b_list', and '_flag_ent_exit'
+    in order to fully form a_list and b_list. The 'a_list' and 'b_list'
+    that it returns are the fully updated vectors of PolyNodes that represent
+    'poly_a' and 'poly_b', respectively. This function also returns
+    'a_idx_list', which at its "ith" index stores the index in 'a_list' at
+    which the "ith" intersection point lies. This function also returns
+    'intr_list', which at the "ith" index contains the cartesian coordinates
+    for the "ith" intersection point. Finally, this function returns 'edges_a'
+    and 'edges_b' which are vectors of the cartesian points of the original
+    polygons, 'poly_a' and 'poly_b'.
+"""
+function _build_ab_list(poly_a, poly_b)
+    # Make a list for nodes of each polygon. Note the definition of PolyNode
+    a_list = Array{PolyNode, 1}(undef, _nedge(poly_a))
+    b_list = Array{PolyNode, 1}(undef, _nedge(poly_b))
+
+    # Initialize arrays to keep track of the important information 
+    # associated with the intersection points of poly_a and poly_b.
+    # I initialize these arrays assumed there will be a maximum of 
+    # 30 intersection points and then I truncate the arrays later.
+    k = 4
+    # intr_list stores the cartesian coordintes of the intersection point
+    intr_list = Array{Tuple{Real, Real}, 1}(undef, k)
```

Comment:
I think we can get rid of this if we add in a `point` field to the PolyNode like I said below.